### ISSUE TITLE
uploader: migrate tasks from legacy BibUpload

### ISSUE THREAD

**Issue by invenio-developers** [inveniosoftware/invenio#1772]
## Idea
Replace the current BibUpload.
## Proposal
Create a new uploader keeping this list of ideas about it in mind:
* Work with JSON natively rather than MARC, although the last could be uploaded as well.
* Run tasks in parallel when possible, for example one huge insert coming from one file could generate several insert tasks that could run in parallel.
* Easy to customise depending on the installation needs. For that we thought about creating *lightweight workflows* (https://github.com/romanchyla/workflow) for each of the common/current modes, so each installation could override the current ones or implement their owns.
* This lightweight workflows should be made out of small monolithic tasks and it would be nice to have a rollback function for each of this functions in case of error in any of them while running the workflow (reverse workflow?)
* Plug the new documents module with a default document storage (same behaviour as BibDocFile).
* Fire signals before and after running any workflow would be a good thing to "connect" the new uploader with other Invenio tools like `bibindex`, `webcoll`, etc.

## Tasks

- [x] insert mode
- [ ] replace mode 
- [ ] correct mode

--
see implementation at #1850

**Comment by jirikuncar** [comment]
This should priority for @CERNDocumentServer. The usage of new *documents* storage as well as migration of *BibDocFiles* to *documents* can be very limited without proper way of updating record using new *uploader*.

**Comment by lnielsen** [comment]
I will look into when starting migration to documents on Zenodo. No promises though ;-)

**Comment by kaplun** [comment]
cc: @jalavik @jmartinm

**Comment by tiborsimko** [comment]
We now have `invenio-records` with its own upload API.
